python/lsst/sims/GalSimInterface/galSimInterpreter.py

```diff
@@ -598,8 +599,11 @@ def write_checkpoint(self, force=False):
                                rng=self._rng,
                                drawn_objects=self.drawn_objects,
                                centroid_objects=self.centroid_list)
-            with open(self.checkpoint_file, 'wb') as output:
-                pickle.dump(image_state, output)
+            with tempfile.NamedTemporaryFile(mode='wb', delete=False,
+                                             dir='.') as tmp:
+                pickle.dump(image_state, tmp)
+                os.fsync(tmp.fileno())
```

Comment:
os.fsync docs say:

"If you’re starting with a Python file object f, first do f.flush(), and then do os.fsync(f.fileno()), to ensure that all internal buffers associated with f are written to disk."

Do we need to to that?

Reply:
yes, I was just putting that in.